src/socket/mod.rs

```diff
@@ -0,0 +1,58 @@
+use bcc_sys::bccapi::bpf_prog_type_BPF_PROG_TYPE_SOCKET_FILTER as BPF_PROG_TYPE_SOCKET_FILTER;
+use std::collections::HashMap;
+use std::iter::Iterator;
+
+use crate::core::BPF;
+use crate::error::BccError;
+
+#[derive(Debug, Default)]
+pub struct Socket {
+    handler: Option<String>,
+    ifaces: Vec<String>,
```

Comment:
Should this really be a HashSet ? What happens if we add the same interface twice?